### ISSUE TITLE
Dreambooth: cosine_with_restarts learning rate schedule won't restart

### ISSUE THREAD

**Issue by zetyquickly** [huggingface/diffusers#1232]
### Describe the bug

Hello,

During log observation of a three dreambooth training logs found that `--lr_scheduler="cosine_with_restarts" lead to a cosine-alike lr change on a tensorboard.

<img width="386" alt="Screen Shot 2022-11-09 at 2 50 45 PM" src="https://user-images.githubusercontent.com/25350960/200959317-3239dc33-4ac9-42bb-8d60-fd46b78f4b9f.png">

### Reproduction

Training finishes successfully and results are pretty good

### System Info

diffusers==0.6.0.dev0

**Comment by patrickvonplaten** [comment]
@patil-suraj could you take a look here? :-)

**Comment by zetyquickly** [comment]
Let me clarify a little, there's no restarts

**Comment by patrickvonplaten** [comment]
@patil-suraj is pretty busy at the moment, could you maybe help us here by providing a reproducible google colab or code snippet?

**Comment by pcuenca** [comment]
@zetyquickly I've checked this issue (sorry for the delay), and the problem is that there is currently no way for the training script to specify the number of cycles used for that schedule, hence it only uses 1 cycle. Most learning rates are created [here](get_cosine_with_hard_restarts_schedule_with_warmup), so the value used for the number of cycles is the default `1`: https://github.com/huggingface/diffusers/blob/main/src/diffusers/optimization.py#LL143C27-L143C27

This affects other schedules too.

We could add additional arguments to the training script, but I'm not sure how important this is for training methods such as Dreambooth. What do you think @patrickvonplaten?

**Comment by zetyquickly** [comment]
I can make a PR, if it's not in your top priority list to do

**Comment by pcuenca** [comment]
That'd be amazing @zetyquickly! Maybe we can include the polynomial `power` too?

**Comment by GeneralAwareness** [comment]
Was this fixed as LoRA is using this as well.

**Comment by zetyquickly** [comment]
check whether this script has num_cycles option. if so then it uses the recent changes

**Comment by zetyquickly** [comment]
https://github.com/huggingface/diffusers/pull/1737

**Comment by GeneralAwareness** [comment]
@zetyquickly I linked this, as you can see from yesterday, to Kohya who wrote their LoRA Dreambooth scripts and I have an option called If lr_scheduler = cosine_with_restarts, update lr_scheduler_num_cycles.

lr_scheduler_num_cycles: 1

I played with it from 1,2,4,8 and all of the results were rubbish in comparison to poly with a power of 4.8.  4 was the best out of the lot but still bad in comparison and worse loss rate than with Poly power 4.8
